launcher/launcher_test.go

```diff
@@ -281,28 +308,45 @@ func (s *LauncherTestSuite) TestWaitForUpdates() {
 				{1 * time.Millisecond, configA},
 			},
 			expected4,
+			beaterMockCreator,
+			false,
+		},
+		{
+			"invalid updates",
+			incomingConfigs{
+				{40 * time.Millisecond, configErr},
+			},
+			nil,
+			errorReloadBeaterCreator(),
+			true,
```

Comment:
I feel this test case needs to be in a different test function, as those are all success flows and this test covers a failure point.
Eventually, the purpose is to make the test more readable.